app/code/Magento/InventorySales/Plugin/Store/WebsiteResourcePlugin.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventorySales\Plugin\Store;
+
+use Magento\Framework\App\ResourceConnection;
+use Magento\InventorySalesApi\Api\Data\SalesChannelInterface;
+
+class WebsiteResourcePlugin
+{
+    /**
+     * @var ResourceConnection
+     */
+    private $resource;
+
+    /**
+     * WebsiteResourcePlugin constructor.
+     * @param ResourceConnection $resource
+     */
+    public function __construct(
+        ResourceConnection $resource
+    ) {
+        $this->resource = $resource;
+    }
+
+    /**
+     * Get code from database
+     * @param \Magento\Store\Model\ResourceModel\Website $resourceModel
+     * @param int $websiteId
+     * @return string
+     * @throws \Magento\Framework\Exception\LocalizedException
+     */
+    private function getCodeFromDatabase(
+        \Magento\Store\Model\ResourceModel\Website $resourceModel,
+        int $websiteId
+    ) {
```

Comment:
Pls, 
1) use return type hinting
2) `$qry` is not clear name for the variable
3) Need to use `$this->resource` instead of `$resourceModel->getConnection()`, 
generally we don't need dependency on `ResourceModel\Website` for obtaining website code
4) Also, need to think about extracting this logic to separate class